spec.md

```diff
@@ -0,0 +1,955 @@
+
+Implementations MAY remove leading and trailing whitespace of individual values of a multi-valued header without changing sematics. Implementations MAY also remove empty values in a multi-valued header without changing semantics.
+
+> [!WARNING]
+>
+> **Backwards-Compatible Change** in version 1.1.0
+>
+> Multi-Valued Headers were introduced in version 1.1.0 of the format.
+> Headers indicated as multi-valued were single-valued in previous versions.
+
+> [!WARNING]
+>
+> **Backwards-Compatible Change** in version ???
+>
+> Concatenating multiple occurrences of multi-valued headers was introduced in version ??? of the format.
+> Previously handling of repeated headers was not covered by this specification.
+
+>  [!CAUTION]
+>
+> The use of multi-valued headers has not been decided yet ([#22](https://github.com/UltraStar-Deluxe/format/issues/22)).
+
+### 3.2. File References
+
+Some headers reference other files, most notably `AUDIO`, `VIDEO`, `COVER`, and `BACKGROUND`.
+These file references are always relative to the textfile from which they are referenced.
+As a security measure file references MUST NOT use absolute paths.
+Implementations SHOULD refuse to load absolutely referenced files.
+
+> [!CAUTION]
+>
+> Whether absolute paths are allowed or not hasn't been decided yet.
+
+> [!IMPORTANT]
+>
+> In Windows file names are case-insensitive (i.e. there cannot be two files in a folder that differ only by their case).
+> Linux and macOS however, use fully case-sensitive file systems.
+> Implementations might need to pay special attention to this fact to ensure that files are compatible across all systems.
+
+### 3.3. The `VERSION` Header
+
+```
+Required:     Yes
+Multi-Valued: No
+Syntax:       <valid semver>
+Since:        1.0.0
+```
+
+The `VERSION` header indicates the version of this specification that a file complies to.
+The value of the tag is a [semantic version](https://semver.org),
+meaning that an implementation supporting version 1.0.0 of this spec should be able to process files using a version of 1.1.0 without any changes (although new features might not be supported).
+Implementations SHOULD NOT attempt to process files with a higher major version than they were designed to work with.
+
+In absence of the `VERSION` header implementations SHOULD assume the version 0.3.0.
+Implementations SHOULD reject a file based on the value of the `VERSION` header,
+in particular if the value is syntactically invalid.
+Applications MAY reject prerelease-versions altogether.
+
+The `VERSION` header SHOULD be the first header in a file.
+
+### 3.4. The `BPM` Header
+
+```
+Required:     Yes
+Multi-Valued: No
+Syntax:       1*DIGIT [ period 1*DIGIT ]
+Since:        0.1.0
+```
+
+The `BPM` header indicates the number of beats per minute.
+The notes in a song are quantized in beats.
+A single beat is the smallest unit of time that can be present in a song.
+The value is a 32 bit floating point number.
```

Comment:
I would remove this line as which data type the parsed number is stored in is a pure implementation details and has nothing to do with the format.

Reply:
I removed the 32-bit part but I left the floating point part for clarity. My intention behind the 32-bit part was that compliant implementations wouldn't have to deal with arbitrarily large values (I think I saw some blog post about JSON technically allowing arbitrarily large numbers). But I can see that this not really an issue for us, practically speaking.

Reply:
But _floating point_ as well is just a way to store numbers. I would just call it a *decimal*.
You could avoid arbitrary scale and precision syntactically: `1*4DIGIT [ period 1*2DIGIT ]`

Reply:
Ah of course, you are right. I also updated the other places where it said "floating point number". I think I'll leave the precision unspecified for now and maybe discuss it in a separate issue later on.